src/Symfony/Component/Serializer/Tests/SerializerTest.php

```diff
@@ -1050,6 +1051,141 @@ public function testCollectDenormalizationErrorsWithConstructor()
+
+            return;
+        }
+    }
+
+    /** @requires PHP 8.0 */
+    public function testCollectDenormalizationErrorsWithConstructorMissingParameters()
+    {
+        $json = '{"xyz": "abc"}';
+
+        $classMetadataFactory = new ClassMetadataFactory(new AnnotationLoader(new AnnotationReader()));
+        $extractor = new PropertyInfoExtractor([], [new PhpDocExtractor(), new ReflectionExtractor()]);
+
+        $serializer = new Serializer(
+            [
+                new ObjectNormalizer($classMetadataFactory, null, null, $extractor, new ClassDiscriminatorFromClassMetadata($classMetadataFactory)),
+            ],
+            ['json' => new JsonEncoder()]
+        );
+
+        try {
+            $serializer->deserialize($json, Php80WithPromotedTypedConstructorAndManyParameters::class, 'json', [
+                DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS => true,
+            ]);
+
+            $this->fail();
+        } catch (PartialDenormalizationException $e) {
+            $this->assertInstanceOf(Php80WithPromotedTypedConstructorAndManyParameters::class, $e->getData());
+            $this->assertCount(3, $e->getErrors());
+
+            $error1 = $e->getErrors()[0];
+            $this->assertInstanceOf(NotNormalizableValueException::class, $error1);
+            $this->assertSame('null', $error1->getCurrentType());
+            $this->assertSame(['string'], $error1->getExpectedTypes());
+            $this->assertSame('foo', $error1->getPath());
+            $this->assertTrue($error1->canUseMessageForUser());
+            $this->assertSame('Failed to create object because the object miss the "foo" property.', $error1->getMessage());
+
+            $error2 = $e->getErrors()[1];
+            $this->assertInstanceOf(NotNormalizableValueException::class, $error2);
+            $this->assertSame('null', $error2->getCurrentType());
+            $this->assertSame(['int'], $error2->getExpectedTypes());
+            $this->assertSame('bar', $error2->getPath());
+            $this->assertTrue($error2->canUseMessageForUser());
+            $this->assertSame('Failed to create object because the object miss the "bar" property.', $error2->getMessage());
+
+            $error3 = $e->getErrors()[2];
+            $this->assertInstanceOf(NotNormalizableValueException::class, $error3);
+            $this->assertSame('null', $error3->getCurrentType());
+            $this->assertSame(['string', 'int'], $error3->getExpectedTypes());
+            $this->assertSame('union', $error3->getPath());
+            $this->assertTrue($error3->canUseMessageForUser());
+            $this->assertSame('Failed to create object because the object miss the "union" property.', $error3->getMessage());
+
+            return;
+        }
+    }
+
+    /** @requires PHP 8.0 */
+    public function testCollectDenormalizationErrorsWithConstructorPartiallyMissingParameters()
```

Comment:
That test and the previous one (`testCollectDenormalizationErrorsWithConstructorMissingParameters`) are kind of testing the same stuff, WDYT?